src/core/api/public_api.ts

```diff
@@ -642,6 +651,71 @@ class Player extends EventEmitter<IPublicAPIEvent> {
 
     const isDirectFile = transport === "directfile";
 
+    if (isDirectFile) {
+      this._priv_mediaElementTrackChoiceManager = new MediaElementTrackChoiceManager({
+        preferredAudioTracks: defaultAudioTrack === undefined ?
+          this._priv_preferredAudioTracks :
+          new BehaviorSubject([defaultAudioTrack]),
+        preferredTextTracks: defaultTextTrack === undefined ?
+          this._priv_preferredTextTracks :
+          new BehaviorSubject([defaultTextTrack]),
+      }, this.videoElement);
+
+      let hasSetInitialAudio = false;
+      let hasSetInitialText = false;
+
+      this._priv_mediaElementTrackChoiceManager
+        .addEventListener("availableTracksChange", ({ type }: { type: string }) => {
+          switch (type) {
+            case "video":
+              this._priv_triggerEventIfChanged("availableVideoTracksChange",
+                                               this.getAvailableVideoTracks());
+              break;
+            case "audio":
+              if (!hasSetInitialAudio) {
+                this._priv_mediaElementTrackChoiceManager?.setInitialAudioTrack();
+                hasSetInitialAudio = true;
+              }
```

Comment:
Hmm i don't really know if the MediaElementTrackChoiceManager is supposed to know that it will be refresh each time we want to load a content. I'm not sure about this.